StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Link/LinkPaymentController.swift

```diff
@@ -0,0 +1,220 @@
+        Task {
+            do {
+                try await present(from: presentingViewController)
+                completion(.success(()))
+            } catch {
+                completion(.failure(error))
+            }
+        }
+    }
+
+    /// Presents the Link payment flow, allowing your customer to pay with Link.
+    /// The flow lets your customer log into or create a Link account, select a valid source of funds, and approve the usage of those funds to complete the purchase. The actual purchase will not occur until you call `confirm(from:)`.
+    /// If this method returns successfully, you can call `confirm(from:)` when you're ready to complete the payment.
+    /// - Note: Once `confirm(from:)` returns successfully, calling this method is an error, as payment/setup intents should not be reused. Until then, you may call this method as many times as is necessary.
+    /// - Parameter presentingViewController: The view controller to present the payment flow from
+    /// - Throws: Either `LinkPaymentController.Error.canceled`, meaning the customer canceled the flow, or an error describing what went wrong.
+    @MainActor
+    @_spi(LinkOnly) public func present(from presentingViewController: UIViewController) async throws -> Void {
+        let linkController: PayWithLinkViewController = try await withCheckedThrowingContinuation { [self] continuation in
+            PaymentSheet.load(clientSecret: intentSecret, configuration: configuration) { result in
+                switch result {
+                case .success(let intent, _, let isLinkEnabled):
+                    guard isLinkEnabled else {
+                        continuation.resume(throwing: LinkPaymentController.Error.unavailable)
+                        return
+                    }
+                    self.intent = intent
+                    let linkController = PayWithLinkViewController(
+                        intent: intent,
+                        configuration: self.configuration,
+                        shouldOfferApplePay: false,
+                        shouldFinishOnClose: true,
+                        shouldPayInFlow: false
+                    )
+                    continuation.resume(returning: linkController)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+
+        linkController.payWithLinkDelegate = self
+        linkController.modalPresentationStyle = UIDevice.current.userInterfaceIdiom == .pad
+            ? .formSheet
+            : .overFullScreen
+
+        defer { linkController.dismiss(animated: true) }
+        _ = try await withCheckedThrowingContinuation { continuation in
+            payWithLinkContinuation = continuation
+            presentingViewController.present(linkController, animated: true)
+        }
```

Comment:
Of all the bridging between `async` and non-`async` worlds, this feels the sketchiest to me. Storing continuations (as is required to accommodate a delegate-based API) makes me uncomfortable, and while `defer` is just standing in for Swift's missing `finally` construct, the impact it has on code flow (especially if more code is ever added below the `await`) makes me uneasy. Thoughts on how this can be improved/altered very welcome!

Reply:
(Forgot to submit this comment before)

Other than adding comments to explain what your intent is here, I don't see anything that could improve it, I agree is sketchy and is the one thing that makes me uneasy about adopting a hybrid concurrency approach, but as long as we document along the way I think it's okay.